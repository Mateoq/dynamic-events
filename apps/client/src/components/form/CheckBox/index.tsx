import React from 'react';
import { AnimatePresence, motion } from 'framer-motion';
import { FaCheck } from 'react-icons/fa6';

export interface CheckBoxProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  isChecked?: boolean;
}

export const CheckBox = React.forwardRef<HTMLInputElement, CheckBoxProps>(
  (props, ref) => {
    const { label, isChecked, ...rest } = props;

    return (
      <label className="flex flex-col">
        {label && (
          <span className="text-xs mb-1 font-semibold uppercase text-gray-900">
            {label}
          </span>
        )}
        <div className="h-7 w-7 border-2 px-2.5 rounded appearance-none cursor-pointer border-gray-700 flex items-center justify-center">
          <input {...rest} ref={ref} type="checkbox" className="hidden" />
          <AnimatePresence>
            {isChecked && (
              <motion.div
                className="h-5 w-5 min-w-5"
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                exit={{ opacity: 0 }}
                transition={{ duration: 0.2 }}
              >
                <FaCheck size="100%" />
              </motion.div>
            )}
          </AnimatePresence>
        </div>
      </label>
    );
  }
);
