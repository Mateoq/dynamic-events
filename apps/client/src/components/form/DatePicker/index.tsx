import React from 'react';
import clsx from 'clsx';
import { FaCalendar } from 'react-icons/fa6';
// import Datepicker, { DatepickerType } from 'react-tailwindcss-datepicker';
import DDatePicker, {
  DatePickerProps as DDatePickerProps
} from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';

export type DatePickerProps = DDatePickerProps & {
  label?: string;
  message?: string;
  error?: boolean;
  large?: boolean;
};

export const DatePicker: React.FC<DatePickerProps> = (props) => {
  const { label, message, error, large, ...rest } = props;
  console.log('VALUE_DATE', rest.selected);

  return (
    <label className="flex flex-col w-fit">
      {label && (
        <span
          className={clsx('text-xs mb-1 font-semibold uppercase', {
            'text-gray-900': !error,
            'text-red-600': error
          })}
        >
          {label}
        </span>
      )}
      <div className="relative w-fit">
        <DDatePicker
          {...rest}
          showIcon
          icon={<FaCalendar className="text-gray-800 text-lg mt-1" />}
          className={clsx(
            'min-w-64 min-h-11 border-2 px-2.5 rounded appearance-none cursor-pointer',
            {
              'border-gray-700': !error,
              'border-red-600': error,
              'w-96': large
            }
          )}
        />
      </div>
      {message && error && (
        <div className="mt-1 min-h-3">
          <p className="text-xs text-red-600">{message}</p>
        </div>
      )}
    </label>
  );
};
