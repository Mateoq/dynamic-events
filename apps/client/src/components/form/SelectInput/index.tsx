import React from 'react';
import { FaChevronDown } from 'react-icons/fa';
import clsx from 'clsx';

export type SelectItem = {
  value: string;
  label: string;
};

export interface SelectInputProps
  extends React.SelectHTMLAttributes<HTMLSelectElement> {
  options: SelectItem[];
  label?: string;
  message?: string;
  error?: boolean;
  shrink?: boolean;
}

export const SelectInput = React.forwardRef<
  HTMLSelectElement,
  SelectInputProps
>((props, ref) => {
  const { options, label, message, error, shrink, ...rest } = props;

  return (
    <label className="flex flex-col w-fit">
      {label && (
        <span
          className={clsx('text-xs mb-1 font-semibold uppercase', {
            'text-gray-900': !error,
            'text-red-600': error
          })}
        >
          {label}
        </span>
      )}
      <div className="relative w-fit cursor-pointer">
        <select
          {...rest}
          ref={ref}
          className={clsx(
            'min-h-11 border-2 px-2.5 rounded appearance-none cursor-pointer',
            {
              'border-gray-700': !error,
              'border-red-600': error,
              'min-w-64': !shrink,
              'w-auto': shrink
            }
          )}
        >
          {options.map((opt, index) => (
            <option key={`option_${opt.value}_${index}`} value={opt.value}>
              {opt.label}
            </option>
          ))}
        </select>
        <div className="h-full w-10 absolute right-0 top-0 flex items-center justify-center">
          <FaChevronDown size="18px" />
        </div>
      </div>
      {message && error && (
        <div className="mt-1 min-h-3">
          <p className="text-xs text-red-600">{message}</p>
        </div>
      )}
    </label>
  );
});
