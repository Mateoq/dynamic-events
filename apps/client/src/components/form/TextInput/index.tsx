import React from 'react';
import clsx from 'clsx';

export interface TextInputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  message?: string;
  error?: boolean;
  large?: boolean;
}

export const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
  (props, ref) => {
    const { label, message, error, large, ...rest } = props;

    return (
      <label className="flex flex-col w-fit">
        {label && (
          <span
            className={clsx('text-xs mb-1 font-semibold uppercase', {
              'text-gray-900': !error,
              'text-red-600': error
            })}
          >
            {label}
          </span>
        )}
        <div className="relative w-fit">
          <input
            {...rest}
            ref={ref}
            className={clsx(
              'min-w-64 min-h-11 border-2 px-2.5 rounded appearance-none',
              {
                'border-gray-700': !error,
                'border-red-600': error,
                'w-96': large
              }
            )}
          />
        </div>
        {message && error && (
          <div className="mt-1 min-h-3">
            <p className="text-xs text-red-600">{message}</p>
          </div>
        )}
      </label>
    );
  }
);
