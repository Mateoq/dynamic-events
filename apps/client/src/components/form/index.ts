export * from './Button';
export * from './SelectInput';
export * from './TextInput';
export * from './DatePicker';
export * from './CheckBox';
