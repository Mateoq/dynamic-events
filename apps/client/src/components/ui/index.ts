export * from './Calendar';
export * from './LoadingIcon';
export * from './Modal';
export * from './Toast';
export * from './EventCard';
