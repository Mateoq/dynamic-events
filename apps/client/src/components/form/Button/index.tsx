import React from 'react';

import { LoadingIcon } from '@/components/ui';

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isLoading?: boolean;
}

export const Button: React.FC<ButtonProps> = (props) => {
  const { isLoading, children, ...rest } = props;

  return (
    <button
      {...rest}
      className="min-w-24 h-10 px-3o flex items-center justify-center bg-gray-950 text-white text-lg hover:bg-gray-800 active:bg-gray-700 transition-colors"
    >
      {isLoading ? <LoadingIcon /> : children}
    </button>
  );
};
