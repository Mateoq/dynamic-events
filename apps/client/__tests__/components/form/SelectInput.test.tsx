import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import { SelectInput, SelectItem } from '@/components/form';

const options: SelectItem[] = [
  { label: 'First', value: 'first' },
  { label: 'Second', value: 'second' }
];

describe('SelectInput', () => {
  it('should match snapshot', async () => {
    const { container } = render(
      <SelectInput
        label="SelectInput Label"
        name="select"
        id="select"
        options={options}
        message="Error"
      />
    );
    const label = await screen.findByText('SelectInput Label');

    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
