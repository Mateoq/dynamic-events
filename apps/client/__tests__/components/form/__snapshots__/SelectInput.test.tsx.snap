// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput should match snapshot 1`] = `
<div>
  <label
    class="flex flex-col w-fit"
  >
    <span
      class="text-xs mb-1 font-semibold uppercase text-gray-900"
    >
      SelectInput Label
    </span>
    <div
      class="relative w-fit cursor-pointer"
    >
      <select
        class="min-h-11 border-2 px-2.5 rounded appearance-none cursor-pointer border-gray-700 min-w-64"
        id="select"
        name="select"
        role="listbox"
      >
        <option
          value="first"
        >
          First
        </option>
        <option
          value="second"
        >
          Second
        </option>
      </select>
      <div
        class="h-full w-10 absolute right-0 top-0 flex items-center justify-center"
      >
        <svg
          fill="currentColor"
          height="18px"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="18px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;
